docs/architecture.rst

```diff
@@ -4,9 +4,35 @@
 Architecture
 ============
 
-Endpoint nodes handle all notification PUT requests, looking up in DynamoDB to
+.. image:: http://mozilla-push-service.readthedocs.io/en/latest/assets/push_architecture.svg
+
+Overview
+========
+
+For Autopush, we will focus on the section in the above diagram in the
+*Autopush* square.
+
+Autopush consists of two types of server daemons:
+
+``autopush`` (connection node)
+
+    Run a connection node. These handle large amounts of user agents (Firefox)
+    using the Websocket protocol.
+
+``autoendpoint`` (endpoint node)
+
+    Run an endpoint node. These provide a :term:`WebPush` HTTP API for
+    :term:`Application Servers <AppServer>` to HTTP POST messages to endpoints.
+
+To have a running Push Service for Firefox, both of these server daemons must
+be running and communicating with the same DynamoDB tables. A local DynamoDB
+can be run or AWS DynamoDB.
+
+Endpoint nodes handle all :term:`Notification` POST requests, looking up in
+DynamoDB to
 see what Push server the UAID is connected to. The Endpoint nodes then attempt
-delivery to the Push server.
+delivery to the Push server. If the UAID is not online, the message may be stored
```

Comment:
Nit: it's not fully clear what "the Push server" means here. How does "connection node" or "UAID" sound?